google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb

```diff
@@ -484,6 +485,72 @@ def dead_letter_max_delivery_attempts= new_dead_letter_max_delivery_attempts
           @resource_name = nil
         end
 
+        ##
+        # A policy that specifies the conditions for dead lettering messages in a subscription.
+        #
+        # Dead lettering is done on a best effort basis. The same message might be dead lettered multiple times.
+        #
+        # If validation on any of the fields fails at subscription creation/updation, the create/update subscription
+        # request will fail.
+        #
+        # @return [DeadLetterPolicy, nil] The dead letter policy for the subscription, or `nil`.
```

Comment:
We should document the semantics of "no" dead letter policy (i.e. what does `nil` mean?)

Reply:
Good idea, I'll see what I can add.

Reply:
Updated.